internal/scorecard/alpha/scorecard.go

```diff
@@ -23,69 +23,68 @@ import (
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/kubernetes"
 
 	"github.com/operator-framework/operator-sdk/pkg/apis/scorecard/v1alpha2"
 	"github.com/operator-framework/operator-sdk/version"
 )
 
+type TestRunner interface {
+	RunTest(context.Context, Test) (*v1alpha2.ScorecardTestResult, error)
+}
+
 type Scorecard struct {
-	Config              Config
-	Selector            labels.Selector
-	BundlePath          string
-	WaitTime            time.Duration
-	Namespace           string
-	ServiceAccount      string
+	Config      Config
+	Selector    labels.Selector
+	BundlePath  string
+	SkipCleanup bool
+	TestRunner  TestRunner
+}
+
+type PodTestRunner struct {
 	bundleConfigMapName string
 	Client              kubernetes.Interface
-	SkipCleanup         bool
+	Namespace           string
+	ServiceAccount      string
+}
+
+type FakePodTestRunner struct {
+	TestConfig Scorecard
+	TestResult *v1alpha2.ScorecardTestResult
+	Error      error
 }
 
 // RunTests executes the scorecard tests as configured
-func (o Scorecard) RunTests() (testOutput v1alpha2.ScorecardOutput, err error) {
+func (o Scorecard) RunTests(ctx context.Context) (testOutput v1alpha2.ScorecardOutput, err error) {
 	tests := o.selectTests()
 	if len(tests) == 0 {
 		testOutput.Results = make([]v1alpha2.ScorecardTestResult, 0)
 		return testOutput, err
 	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), o.WaitTime)
-	defer cancel()
-
-	bundleData, err := o.getBundleData()
-	if err != nil {
-		return testOutput, fmt.Errorf("error getting bundle data %w", err)
-	}
-
-	err = o.CreateConfigMap(ctx, bundleData)
-	if err != nil {
-		return testOutput, fmt.Errorf("error creating ConfigMap %w", err)
-	}
-
 	testOutput.Results = make([]v1alpha2.ScorecardTestResult, len(tests))
 
 	for idx, test := range tests {
-		result, err := o.runTest(ctx, test)
+		result, err := o.TestRunner.RunTest(ctx, test)
 		if err != nil {
 			result = convertErrorToResult(test.Name, test.Description, err)
 		}
-		testOutput.Results[idx] = result
+		testOutput.Results[idx] = *result
 	}
 
+	/**
 	if !o.SkipCleanup {
-		defer func() {
-			err := o.deletePods(ctx)
-			if err != nil {
-				testOutput.Results = append(testOutput.Results,
-					convertErrorToResult("", "", err))
-			}
-			err = o.deleteConfigMap(ctx)
-			if err != nil {
-				testOutput.Results = append(testOutput.Results,
-					convertErrorToResult("", "", err))
-			}
-		}()
+		err := o.TestRunner.deletePods(ctx)
+		if err != nil {
+			return testOutput, err
+		}
+		err = o.TestRunner.deleteConfigMap(ctx)
+		if err != nil {
+			return testOutput, err
+		}
```

Comment:
why we need to keep these code if it is commented?

Reply:
comments removed.